### DISCUSSION TITLE
Add an escape hatch for ETH_RPC_URL not to be used as FORK_URL

### DISCUSSION THREAD

**Discussion by PaulRBerg** [foundry-rs/foundry#1233] [General]
The latest nightly release (2022-04-08) ([9c2469](https://github.com/gakonst/foundry/commit/9c2469488c6872e5d17198555f7d8e1a80173151)) introduced a change in the way the `ETH_RPC_URL` environment variable is treated. It seems like when this is enabled, Foundry will now also use it as an alias for `FORK_URL`.

But it looks like there's no escape hatch. We can't keep `ETH_RPC_URL` as an environment variable without automatically having it used as a fork URL also. In certain cases, such as when my Internet connection is poor, I would prefer not to test my contracts on a remote RPC. But I would like to keep `ETH_RPC_URL` as a global environment variable on my machine, so I can use the `cast` CLI.

Thus here's my feature request: let us override this new default behavior. I think the simplest way would be to give `FORK_URL` the utmost priority: if that field is set, and it is set to the empty string, then `ETH_PRC_URL` should not be used as a fork URL anymore.

Example:

```bash
export ETH_RPC_URL="https://rinkeby.infura.io/v3/f0c8e45d48b34a8da606303a83df4d0b"
export FORK_URL=""
```

With these environment variables, Foundry should NOT use the RPC url for running the tests against a remote chain fork.

Alternatively, there could be a new configuration option in `foundry.toml` which provides an explicit escape hatch for this behavior.

**Comment by onbjerg** [comment]
I don't think `FORK_URL` is valid right now. I'm pinging @mds1, I think you've wanted `ETH_RPC_URL` to be valid for some time, and I know for sure others have.

Also pinging @mattsse @gakonst to see if they have some ideas/thoughts

**Comment by mattsse** [comment]
I've been bitten be something similar in the past, it's easy to forget that you set an env var previously and wonder why it takes so long suddenly.

so what we need to decide is

* should we pick up `ETH_RPC_URL` (aliases rn are `FOUNDRY_ETH_RPC_URL` and `DAPP_ETH_RPC_URL`) like we do now?

if yes
* how to opt out?

I think the easiest would indeed be an additional env flag that basically ignores forking?

**Comment by onbjerg** [comment]
Just removed this behavior, so this is not the case after the next nightly

**Reply by PaulRBerg** [reply]
Great!

**Reply by mds1** [reply]
I'm still not sure exactly what problem this was intended to solve. It sounds like the issue was people didn't want to type out long CLI commands, but in the case a better solution would be to configure `foundry.toml` properly and add better support for env vars in the config file as discussed #1034 

(also @onbjerg from https://github.com/gakonst/foundry/discussions/1233#discussioncomment-2530689 it sounds like you think I also originally wanted this feature, so sorry for being unclear there—I'm assuming you got that impression from #1034)

**Comment by gakonst** [comment]
Making `Url` -> `Option<Url>` solves this I think?

```
export ETH_RPC_URL=<..> # or FOUNDRY_FORK_URL
forge t # does not fork
forge t --rpc-url # forks with the env var, errors if the env var is none
forge t --rpc-url <..> # works with whatever arg was provided
```
